src/Service.Host/client/src/components/CartonType.js

```diff
@@ -38,6 +39,10 @@ function CartonType({
         }
     });
 
+    useEffect(() => {
+        initialise({ cartonTypeId });
+    }, [cartonTypeId]);
+
```

Comment:
@richardiphillips @atcampbell 

A simple effect that watches the entities id and calls the initialise method when it changes, as a replacement for the old initialiseOnMount() way of doing things.

 In practicee the id should never change  and this effect only runs on mount, so we could just pass an empty array for the effect to watch, but Dan the man seems to want us to explicitly tell UseEffect() which variables it depends on.

I think if we changed this to an empty array the new linting rules would complain anyway.

There is also potential to extract this out to be a common hook for all components, but its simple enough to just do per component for now.

What do you think?